core/src/main/java/org/elasticsearch/index/similarity/SimilarityService.java

```diff
@@ -138,16 +138,6 @@ Similarity getDefaultSimilarity() {
         }
 
         @Override
-        public float coord(int overlap, int maxOverlap) {
```

Comment:
I think you should change the constructor a few lines above in this file to extend PerFieldSimilarityWrapper(defaultSimilarity) rather than the deprecated default constructor?